File: web-app/online-warehouse/src/app/consignment-note/dto/update-consignment-note-dto.ts
import {CounterpartyDto} from "../../counterparty/dto/counterparty.dto";
import {CarrierDto} from "../../carrier/dto/carrier.dto";
import {ConsignmentNoteGoodsDto} from "./consignment-note-goods-dto";
import {ConsignmentNoteType} from "./enum/consignment-note-type.enum";
import {DriverDto} from "../../carrier/driver/driver.dto";

export class UpdateConsignmentNoteDto {
  id: number;
  number: string;
  shipment: Date;
  counterpartyId: number;
  carrierId: number;
  driverId: number;
  vehicleNumber: string;
  consignmentNoteGoodsList: Array<ConsignmentNoteGoodsDto>;
  consignmentNoteType: ConsignmentNoteType;
  description: string;
}

File: core/src/main/java/by/itechart/consignmentnote/dto/UpdateConsignmentNoteDto.java
package by.itechart.consignmentnote.dto;

import by.itechart.consignmentnote.enums.ConsignmentNoteType;
import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.datatype.jsr310.deser.LocalDateDeserializer;
import com.fasterxml.jackson.datatype.jsr310.ser.LocalDateSerializer;
import lombok.Data;
import org.springframework.format.annotation.DateTimeFormat;

import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.PastOrPresent;
import java.time.LocalDate;
import java.util.List;

@Data
public class UpdateConsignmentNoteDto {
    private Long id;
    @NotEmpty
    private String number;
    @DateTimeFormat(iso = DateTimeFormat.ISO.DATE)
    @JsonSerialize(using = LocalDateSerializer.class)
    @JsonDeserialize(using = LocalDateDeserializer.class)
    @NotNull
    @PastOrPresent
    private LocalDate shipment;
    @NotNull
    private Long counterpartyId;
    @NotNull
    private Long carrierId;
    private Long driverId;
    @NotEmpty
    private String vehicleNumber;
    @JsonBackReference
    @NotEmpty
    private List<ConsignmentNoteGoodsDto> consignmentNoteGoodsList;
    @NotNull
    private ConsignmentNoteType consignmentNoteType;
    private String description;
}
